lib/datasource/gitlab-releases/index.ts

```diff
@@ -0,0 +1,46 @@
+import { cache } from '../../util/cache/package/decorator';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import type { GitlabRelease } from './types';
+
+export class GitlabReleasesDatasource extends Datasource {
+  static readonly id = 'gitlab-releases';
+
+  readonly defaultRegistryUrls = ['https://gitlab.com'];
+
+  static readonly registryStrategy = 'first';
+
+  constructor() {
+    super(GitlabReleasesDatasource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${GitlabReleasesDatasource.id}`,
+    key: ({ registryUrl, lookupName }: GetReleasesConfig) =>
+      `${registryUrl}/${lookupName}`,
+  })
+  async getReleases({
+    registryUrl,
+    lookupName,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const urlEncodedRepo = encodeURIComponent(lookupName);
+    const apiUrl = `${registryUrl}/api/v4/projects/${urlEncodedRepo}/releases`;
+
+    const gitlabReleasesResponse = (
+      await this.http.getJson<GitlabRelease[]>(apiUrl)
+    ).body;
```

Comment:
Can you add some error handling here? `handleGenericErrors` will handle 429s and 5xxs for you.